File: Using TDD with ASP.NET MVC/[VB]-Using TDD with ASP.NET MVC/VB/MvcContacts/Models/IContactRepository.vb
' <snippet2>
Public Interface IContactRepository
    Sub CreateNewContact(ByVal contactToCreate As Contact)
    Sub DeleteContact(ByVal id As Integer)
    Function GetContactByID(ByVal id As Integer) As Contact
    Function GetAllContacts() As IEnumerable(Of Contact)
    Function SaveChanges() As Integer
End Interface
' </snippet2>

File: Using TDD with ASP.NET MVC/[C#]-Using TDD with ASP.NET MVC/C#/MvcContacts/Models/IContactRepository.cs
// <snippet2>
using System;
using System.Collections.Generic;

namespace MvcContacts.Models {
    public interface IContactRepository {
        void CreateNewContact(Contact contactToCreate);
        void DeleteContact(int id);
        Contact GetContactByID(int id);
        IEnumerable<Contact> GetAllContacts();
        int SaveChanges();

    }
} 
// </snippet2>
